### PR TITLE
Bootstrap frontend

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Add first POC React-based frontend component.

## Proposal

- [x] add base frontend stack (turbo / next.js / E-Charts)
- [x] add `DailyViews` component
- [x] integrate frontend quality checks in the CI
- [x] update documentation

### CODE REVIEW

File: src/frontend/packages/ui/video/Views.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React, { useState, useEffect } from "react";
+import ReactECharts from "echarts-for-react";
+import cloneDeep from "lodash.clonedeep";
+
+const warrenBackendRootURL: string = "http://localhost:8100";
```
Comment (jmaupetit) [inline src/frontend/packages/ui/video/Views.tsx:5]: Should be a global configuration.

File: src/frontend/packages/ui/video/Views.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+type DailyViews = {
+  day: string;
+  views: number;
+};
+
+export const DailyViews = () => {
+  const videoId = "uuid://8d386f48-3baa-4acf-8a46-0f2be4ae243e";
```
Comment (jmaupetit) [inline src/frontend/packages/ui/video/Views.tsx:13]: Should be send via component props.

File: src/frontend/packages/ui/video/Views.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    // FIXME
+    // Handle fetch errors
+    fetch(`${warrenBackendRootURL}/api/v1/video/${videoId}/views`)
+      .then((response) => response.json())
+      .then((response) => {
+        const daily_views: Array<DailyViews> = response.daily_views;
+        newOption.xAxis.data = daily_views.map((d) => d.day);
+        newOption.series[0].data = daily_views.map((d) => d.views);
+        setOption(newOption);
+      });
```
Comment (jmaupetit) [inline src/frontend/packages/ui/video/Views.tsx:51]: Handle fetch errors.

File: src/frontend/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "lint": "turbo run lint",
+    "format": "prettier --write \"**/*.{ts,tsx,md}\""
+  },
+  "devDependencies": {
+    "eslint-config-custom": "*",
+    "prettier": "latest",
+    "turbo": "latest"
+  },
+  "engines": {
+    "node": ">=14.0.0"
```
Comment (lunika) [inline src/frontend/package.json:21]: on a fresh new project, you can start with version 16. Unless one of your dependency requires node 14

File: src/frontend/packages/ui/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import * as React from "react";
+export const Button = () => {
+  return <button>Boop</button>;
```
Comment (lunika) [inline src/frontend/packages/ui/Button.tsx:3]: You can probably use cunningham for this kind of component ?

File: src/frontend/packages/ui/video/Views.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  day: string;
+  views: number;
+};
+
+type DailyViewsProps = {
+  videoId: string;
+};
+
+export const DailyViews = (props: DailyViewsProps) => {
+  const { videoId } = props;
```
Comment (lunika) [inline src/frontend/packages/ui/video/Views.tsx:20]:
```suggestion
```

File: src/frontend/packages/ui/video/Views.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+type DailyViewsResponseItem = {
+  day: string;
+  views: number;
+};
+
+type DailyViewsProps = {
+  videoId: string;
+};
+
+export const DailyViews = (props: DailyViewsProps) => {
```
Comment (lunika) [inline src/frontend/packages/ui/video/Views.tsx:19]:
```suggestion
export const DailyViews = ({ videoId }: DailyViewsProps) => {
```

File: src/frontend/packages/ui/video/Views.tsx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+        )
+        .then((res) => res.data)
+        .then((data) => {
+          const newOption = cloneDeep(option);
+          const daily_views: Array<DailyViewsResponseItem> = data.daily_views;
+          newOption.xAxis.data = daily_views.map((d) => d.day);
+          newOption.series[0].data = daily_views.map((d) => d.views);
+          setOption(newOption);
+          return data;
+        }),
```
Comment (lunika) [inline src/frontend/packages/ui/video/Views.tsx:70]: IMO all of this should be managed in the `onSuccess` of `react-query`. the `queryFn` is just responsible of fetching data, not applying treatment on it.

File: src/frontend/README.md [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+This turborepo uses [Yarn](https://classic.yarnpkg.com/) as a package manager. It includes the following packages/apps:
+
+### Apps and Packages
+
+- `docs`: a [Next.js](https://nextjs.org/) app
+- `web`: another [Next.js](https://nextjs.org/) app
+- `ui`: a stub React component library shared by both `web` and `docs` applications
+- `eslint-config-custom`: `eslint` configurations (includes `eslint-config-next` and `eslint-config-prettier`)
+- `tsconfig`: `tsconfig.json`s used throughout the monorepo
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/frontend/README.md:15]: AFAIU we use eslint by default (and prettier).

[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/frontend/README.md:15]: After thought: your advice is to remove tslint-related configuration to focus on eslint only?

File: src/frontend/package.json (hunk 4 above)
Comment (jmaupetit) [reply src/frontend/package.json:21]: done

File: src/frontend/packages/ui/Button.tsx (hunk 5 above)
Comment (jmaupetit) [reply src/frontend/packages/ui/Button.tsx:3]: That's precisely the idea! I've removed this component and added our `DailyViews` component to the docs. Still need to write it.

File: src/frontend/packages/ui/video/Views.tsx (hunk 8 above)
Comment (jmaupetit) [reply src/frontend/packages/ui/video/Views.tsx:70]: done

File: src/frontend/packages/ui/Button.tsx
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/frontend/packages/ui/Button.tsx:1]: I did.

File: src/frontend/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (lunika) [reply src/frontend/README.md:15]: Yes you can remove tslint and focus only on eslint. Take a look of the configuration of eslint we have on Marsha or or Cunningham.

File: src/frontend/apps/web/pages/index.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+
+import { DailyViews } from "ui";
+
+const queryClient = new QueryClient();
```
Comment (SergioSim) [inline src/frontend/apps/web/pages/index.tsx:6]:
```suggestion
const queryClient = new QueryClient({
  defaultOptions: {
  queries: {
    refetchOnWindowFocus: false, // default: true
  },
  }
});
```
The default behavior of the QueryClient is to re-fetch the query on each change of focus, is this the expected behavior?

File: src/frontend/apps/web/README.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+```bash
+yarn dev
+```
+
+Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
+
+You can start editing the page by modifying `pages/index.js`. The page auto-updates as you edit the file.
+
+[API routes](https://nextjs.org/docs/api-routes/introduction) can be accessed on [http://localhost:3000/api/hello](http://localhost:3000/api/hello). This endpoint can be edited in `pages/api/hello.js`.
```
Comment (SergioSim) [inline src/frontend/apps/web/README.md:13]: `api/hello.js` seems to be missing?

**Review by SergioSim — COMMENTED** [review]
This looks great) :+1: 
Have only minor suggestions)

File: src/frontend/packages/ui/video/Views.tsx [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+  day: string;
+  views: number;
+};
+
+type DailyViewsProps = {
+  videoId: string;
+};
+
+export const DailyViews = ({ videoId }: DailyViewsProps) => {
+  const queryClient = useQueryClient();
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply src/frontend/packages/ui/video/Views.tsx:20]: Good catch! Looks like a copy/pasta.

File: src/frontend/apps/web/pages/index.tsx (hunk 10 above)
Comment (jmaupetit) [reply src/frontend/apps/web/pages/index.tsx:6]: I was also surprised by this behavior. You are right: I have changed the defaults.

File: src/frontend/apps/web/README.md (hunk 11 above)
Comment (jmaupetit) [reply src/frontend/apps/web/README.md:13]: This whole documentation is not relevant. I'll re-write it when the app is more advanced.

File: src/frontend/apps/web/pages/index.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
+import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
+
+import { DailyViews } from "ui";
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false, // default: true
```
Comment (jmaupetit) [inline src/frontend/apps/web/pages/index.tsx:9]:
```suggestion
      refetchOnWindowFocus: false,
```

**Pull request merged**